orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/monitoreddeploy/NotifyDeployCompletedTask.java

```diff
@@ -40,13 +40,33 @@
 
   @Override
   public @Nonnull TaskResult executeInternal(
-      Stage stage,
-      MonitoredDeployStageData context,
-      DeploymentMonitorDefinition monitorDefinition) {
-    // TODO(mvulfson): actually populate the request data
+      Stage stage, DeploymentMonitorDefinition monitorDefinition) {
     DeploymentCompletedRequest request = new DeploymentCompletedRequest(stage);
-    monitorDefinition.getService().notifyCompleted(request);
 
+    request.setStatus(
+        convertStageStatus(
+            (boolean) stage.getContext().getOrDefault("hasDeploymentFailed", false)));
+    request.setRollback(DeploymentCompletedRequest.DeploymentStatus.ROLLBACK_NOT_PERFORMED);
+
+    // check whether rollback was initiated and successful
+    if (stage.getParent() != null) {
+      stage.getParent().directChildren().stream()
+          .filter(s -> s.getType().equals(RollbackClusterStage.PIPELINE_CONFIG_TYPE))
+          .findFirst()
+          .ifPresent(
+              foundRollbackStage ->
+                  request.setRollback(
+                      convertStageStatus(
+                          foundRollbackStage.getStatus() != ExecutionStatus.SUCCEEDED)));
+    }
+
+    monitorDefinition.getService().notifyCompleted(request);
     return TaskResult.ofStatus(ExecutionStatus.SUCCEEDED);
   }
+
+  private DeploymentCompletedRequest.DeploymentStatus convertStageStatus(Boolean failedStatus) {
```

Comment:
nit: you can make this boolean now
```suggestion
  private DeploymentCompletedRequest.DeploymentStatus convertStageStatus(boolean failedStatus) {
```